01_git/scissors/game.cpp

```diff
@@ -0,0 +1,149 @@
+#include <cstdlib>
+#include <iostream>
+
+// Not using enum class since too lazy to
+// do explicit casts to size_t everywhere.
+enum Choice
+{
+    SCISSORS = 0,
+    PAPER,
+    ROCK,
+    CHOICE_LAST
+};
+
+enum GameResult
+{
+    PLAYER_WINS = 0,
+    CPU_WINS,
+    DRAW,
+    GR_LAST
+};
+
+namespace
+{
+
+const char* gaItemsStrRepr[]
+{
+    "scissors", // SCISSORS
+    "paper",    // PAPER
+    "rock"      // ROCK
+};
+
+const char* gaGameResStrRepr[]
```

Comment:
Datatypes better to move to the header file

Reply:
That makes sense in case the data types are used in more than one module (unit) but this task is too simple to be split into several files. In case this change is mandatory I may introduce custom type aliases to be exported.

Reply:
No that is not mandatory, it is just a habit to separate data and code in the current project it may not have an influence. It is a recommendation.